ykrt/src/compile/jitc_yk/codegen/x64/lsregalloc.rs

```diff
@@ -293,6 +293,14 @@ impl LSRegAlloc<'_> {
         iidx: InstIdx,
         constraints: [RegConstraint<Rq>; N],
     ) -> [Rq; N] {
+        // No constraint operands should be float-typed.
+        #[cfg(debug_assertions)]
```

Comment:
I think this can be something like `debug_assert!(constraints.iter().find(|x| self.m.type(...) = ...).is_none());`?

Reply:
Do you mind if we leave it as-is? If we do a one-liner of combinators it becomes hard to understand.

The change looks something like:
```diff
iff --git a/ykrt/src/compile/jitc_yk/codegen/x64/lsregalloc.rs b/ykrt/src/compile/jitc_yk/codegen/x64/lsregalloc.rs
index 7f62501b..45093bab 100644
--- a/ykrt/src/compile/jitc_yk/codegen/x64/lsregalloc.rs
+++ b/ykrt/src/compile/jitc_yk/codegen/x64/lsregalloc.rs
@@ -294,12 +294,13 @@ impl LSRegAlloc<'_> {
         constraints: [RegConstraint<Rq>; N],
     ) -> [Rq; N] {
         // No constraint operands should be float-typed.
-        #[cfg(debug_assertions)]
-        for c in &constraints {
-            if let Some(o) = c.operand() {
-                debug_assert!(!matches!(self.m.type_(o.tyidx(self.m)), Ty::Float(_)));
-            }
-        }
+        debug_assert!(constraints
+            .iter()
+            .find(|c| matches!(
+                c.operand().map(|o| self.m.type_(o.tyidx(self.m))),
+                Some(Ty::Float(_))
+            ))
+            .is_none());
```

(this still isn't quite right)

Reply:
OK.